### PR TITLE
Sequtils improvements

### PR DESCRIPTION
**Pull request by narimiran**
- one letter parameter names
- parameter types are openArray where possible
- added missing tests and new tests for arrays as parameters

### CODE REVIEW

File: lib/pure/collections/sequtils.nim [hunk 1]
```diff
@@ -43,8 +43,8 @@ proc concat*[T](seqs: varargs[seq[T]]): seq[T] =
       result[i] = itm
       inc(i)
 
-proc count*[T](list: seq[T], item: T): int =
-  ## Count the occurrences of the item `item` in the sequence `list`.
+proc count*[T](s: openArray[T], x: T): int =
+  ## Count the occurrences of the item `x` in the sequence `s`.
```
Comment (Araq) [inline lib/pure/collections/sequtils.nim]: But it's not a sequence anymore.

Comment (narimiran) [reply lib/pure/collections/sequtils.nim]: Container?

Comment (Araq) [reply lib/pure/collections/sequtils.nim]: my point is that the comment is wrong.

Comment (narimiran) [reply lib/pure/collections/sequtils.nim]: I understood that :) I'm asking what word should be used in comments to better describe it. Is it "containter" or do you prefer something else?

Comment (Araq) [reply lib/pure/collections/sequtils.nim]: container is fine.

Comment (narimiran) [reply lib/pure/collections/sequtils.nim]: A new commit submitted, changes this to "container", and makes some other minor documentation fixes.

**Comment by Araq** [comment]
The "container s"? For openArray I would have used ``a``. But I'll merge this now, no need to spend even more time on this...

**Comment by genotrance** [comment]
Part of this PR fixes #3801.

**Comment by dom96** [comment]
What is the rationale for making the parameters single letters?!

**Pull request merged**